### PR TITLE
feat: Added binary to octal conversion

### PR DESCRIPTION
**Pull request by peterlucchino**
#### Description of Change
Added a function which converts binary strings to octal strings

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
This is my first open source github contribution, any feedback would be appreciated. 
If any part of this doesn't work or is incorrect feel free to tell me what I did wrong or could have done better

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1934"></a>

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thanks for submitting a PR! :smile: Please remove all the other files and keep the `others/binary_to_octal.cpp` file. Let us know if you need any help. Thanks.

**Comment by peterlucchino** [comment]
Should I just submit a new pull request? I'm not sure how to remove files from this one

File: others/binary_to_octal.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            if (f[q] == '1') {
+                count += 1;
+            }
+        }
+        output += std::to_string(count);
+        count = 0;
+    }
+    return output;
+}
+
```
Comment (Qannaf) [inline others/binary_to_octal.cpp:37]:
```suggestion
void test()
{
 assert(binary_to_octal("111101") == "75");
    assert(binary_to_octal("111101101110100010110011011") == "755642633");
    assert(binary_to_octal("1") == "1");
    assert(binary_to_octal("10") == "2");
    assert(binary_to_octal("1000101010") == "1052");
}
```

File: others/binary_to_octal.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+int main() {
+    assert(binary_to_octal("111101") == "75");
+    assert(binary_to_octal("111101101110100010110011011") == "755642633");
```
Comment (Qannaf) [inline others/binary_to_octal.cpp:40]: test();

File: others/binary_to_octal.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    assert(binary_to_octal("1") == "1");
+    assert(binary_to_octal("10") == "2");
+    assert(binary_to_octal("1000101010") == "1052");
```
Comment (Qannaf) [inline others/binary_to_octal.cpp:43]: test();

**Comment by CarlosZoft** [comment]
@peterlucchino 
if you decide to create a new pull request, close the current pull request and submit it with the requested changes.

**Pull request closed without merging**